lib/graphql/schema/build_from_definition.rb

```diff
@@ -311,16 +325,18 @@ def build_fields(owner, field_definitions, type_resolver, default_resolve:)
               method_conflict_warning: false,
               camelize: false,
               resolver_method: resolve_method_name,
-            ) do
-              builder.build_arguments(self, field_definition.arguments, type_resolver)
-
-              # Don't do this for interfaces
-              if default_resolve
-                owner.send(:define_method, resolve_method_name) do |**args|
-                  field_instance = self.class.get_field(field_definition.name)
-                  default_resolve.call(field_instance, object, args, context)
+            )
+
+            builder.build_arguments(schema_field_defn, field_definition.arguments, type_resolver)
+
+            # Don't do this for interfaces
+            if default_resolve
+              owner.class_eval <<-RUBY, __FILE__, __LINE__
+                def #{resolve_method_name}(**args)
+                  field_instance = self.class.get_field("#{field_definition.name}")
+                  context.schema.definition_default_resolve.call(self.class, field_instance, object, args, context)
                 end
```

Comment:
I think you just hit a common gotcha here. The file's `frozen_string_literal: true` isn't inherited when you do `class_eval`.

So you can save some allocations by adding it inside the `eval` as well. I think there was a couple other eval that could benefit from it.

Reply:
Oh, thank you! I could see allocations pointing to these lines, but I couldn't figure out _why_. I'll add that and give it another go, thanks.